### PR TITLE
Fixed grunt master to work with submodules correctly

### PR DESCRIPTION
**Pull request by allouis**
refs #9441

This should bring `grunt master` command in line with how it should work.

It should only error if there are changes to tracked files which have not been committed - this ensures no work is lost.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -215,6 +215,9 @@ const configureGrunt = function (grunt) {
                     var upstream = grunt.option('upstream') || process.env.GHOST_UPSTREAM || 'upstream';
                     grunt.log.writeln('Pulling down the latest master from ' + upstream);
                     return `
+                        git submodule sync
+                        git submodule update
```
Comment (allouis) [inline Gruntfile.js]: This makes sure that if it's just the submodule being out of date then it doesn't get caught in the `git diff` check

File: Gruntfile.js [hunk 2]
```diff
@@ -218,7 +218,7 @@ const configureGrunt = function (grunt) {
                         git submodule sync
                         git submodule update
 
-                        if ! git diff --exit-code --quiet; then
+                        if ! git diff --exit-code --quiet --ignore-submodules=untracked; then
```
Comment (allouis) [inline Gruntfile.js]: This means that a submodule will not come up dirty if there are untracked files in there - only if there are uncommitted  changes to tracked files

**Review by allouis — COMMENTED** [review]
@ErisDS Could you take a look at this please? I've run this locally with changes to submodules in three ways.

1) `echo "whatever" > content/themes/casper/random-file.txt`
2) `git submodule foreach "git checkout HEAD~~~"`
3) `echo "changes" >> content/themes/casper/default.hbs`

Only 3) errors for me - is that correct?

**Comment by ErisDS** [comment]
This sounds good to me 👍

**Pull request merged**